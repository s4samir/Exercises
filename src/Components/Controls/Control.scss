.controls{
    height: 3em;
    display: flex;
    justify-content: center;
    .timer{
        margin: 1em;
        min-width: 5em;
    }
    .newGame{
        display: flex;
        place-self: center;
    }
    .timer_input{
        width: 2em;
    }
}