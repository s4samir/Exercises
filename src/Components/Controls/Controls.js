/*
 * Authored by Samir Kishore
 */
import React from "react";
import PropTypes from "prop-types";
import Button from "@material-ui/core/Button";
import "./Control.scss";
import TimerWatch from "./TimerWatch/TimerWatch";

export function Controls({ startTheTimer, onNewGame }) {
  return (
    <div className="controls">
      <TimerWatch startTimer={startTheTimer} />
      <div className="newGame">
        <Button variant="contained" color="primary" onClick={onNewGame}>
          New Game
        </Button>
      </div>
    </div>
  );
}

Controls.propTypes = {
  startTheTimer: PropTypes.bool.isRequired,
  onNewGame: PropTypes.func.isRequired,
};
