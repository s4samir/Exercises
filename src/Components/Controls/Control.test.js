/*
 * Authored by Samir Kishore
 */
import React from "react";
import { create } from "react-test-renderer";
import Controls from "./Controls";

describe("Controls", () => {
  it("creates a snapshot of Control", () => {
    const props = {
      startTheTimer: false,
      onNewGame: () => {},
    };
    const wrapper = create(<Controls {...props} />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
