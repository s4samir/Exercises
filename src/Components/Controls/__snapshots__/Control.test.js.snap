// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls creates a snapshot of Control 1`] = `
<div
  className="controls"
>
  <div
    className="timer"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-flex-end"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 1H9v2h6V1zm-4 13h2V8h-2v6zm8.03-6.61l1.42-1.42c-.43-.51-.9-.99-1.41-1.41l-1.42 1.42C16.07 4.74 14.12 4 12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9 9-4.03 9-9c0-2.12-.74-4.07-1.97-5.61zM12 20c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
          />
        </svg>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormControl-root MuiTextField-root timer_input"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
              disabled={true}
              id="input-with-icon-grid"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value={0}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="newGame"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        New Game
      </span>
    </button>
  </div>
</div>
`;
